pkg/driver/controller_modify_volume.go

```diff
@@ -187,52 +205,78 @@ func (d *controllerService) ModifyVolumeProperties(
 		return nil, err
 	}
 
+	options, err := parseModifyVolumeParameters(req.GetParameters())
+	if err != nil {
+		return nil, err
+	}
+
+	name := req.GetName()
+	err = d.modifyVolumeWithCoalescing(ctx, name, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return &rpc.ModifyVolumePropertiesResponse{}, nil
+}
+
+func validateModifyVolumePropertiesRequest(req *rpc.ModifyVolumePropertiesRequest) error {
 	name := req.GetName()
-	modifyOptions := cloud.ModifyDiskOptions{}
-	for key, value := range req.GetParameters() {
+	if name == "" {
+		return status.Error(codes.InvalidArgument, "Volume name not provided")
+	}
+	return nil
+}
+
+func parseModifyVolumeParameters(params map[string]string) (*cloud.ModifyDiskOptions, error) {
+	options := cloud.ModifyDiskOptions{}
+
+	for key, value := range params {
 		switch key {
 		case ModificationKeyIOPS:
 			iops, err := strconv.Atoi(value)
 			if err != nil {
 				return nil, status.Errorf(codes.InvalidArgument, "Could not parse IOPS: %q", value)
 			}
-			modifyOptions.IOPS = iops
+			options.IOPS = iops
 		case ModificationKeyThroughput:
 			throughput, err := strconv.Atoi(value)
 			if err != nil {
 				return nil, status.Errorf(codes.InvalidArgument, "Could not parse throughput: %q", value)
 			}
-			modifyOptions.Throughput = throughput
+			options.Throughput = throughput
+		case DeprecatedModificationKeyVolumeType:
+			if _, ok := params[ModificationKeyVolumeType]; ok {
+				klog.Infof("Ignoring deprecated key `volumeType` because preferred key `type` is present")
+				continue
+			}
+			klog.InfoS("Key `volumeType` is deprecated, please use `type` instead")
```

Comment:
Can we explicitly define verbosity here? e.g. `klog.V(4)`

Reply:
I'm partially against this always being logged at level 1 unless we intend to actually remove `volumeType` eventually because it will spam cx logs. 

But to be fair we have the same log level for [deprecated fsType parameter](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/6f67b3d2f78fe10c95b9beeeb32aa88ffd58038a/pkg/driver/controller.go#L160) so V1 is consistent with that and we can always raise a separate issue/PR.